### ISSUE TITLE
Cookies

### ISSUE THREAD

**Issue by Shadow53** [Shadow53/student-queue#11]
Basically, if I refresh the teacher or admin pages, I have to type the password in again. I know nothing of setting cookies or the like, but perhaps there would be a way of providing authentication cookies for that particular page.

Plus: if the cookie for the queue admin page authenticates you for the teacher page as well, but not the other way around.

**Comment by Shadow53** [comment]
Okay, so I've had a bit of time to look into this and I found [this library](https://www.npmjs.com/package/cookies) that allows cookie signing with [this one](https://www.npmjs.com/package/keygrip). It looks like it'll be some work to implement, especially considering I know nothing about managing cookies.

**Comment by Shadow53** [comment]
Turns out Express has its own [cookie parser](https://github.com/expressjs/cookie-parser) library it bundles with. It would be better to use that than to throw in yet more dependencies, so I'm going to try using that first.

**Comment by Shadow53** [comment]
After looking at the cookie parser library and the response.cookie() API [here](http://expressjs.com/en/4x/api.html#res.cookie) I have realized that I will need to rework how the whole login method works. The way to set a cookie is on a response to a request made by the browser. I should be able to make the modal make a request using $.ajax and, if the response contains a cookie, that should work.

Further work has found that I will need to (and really should) replace all of the communication with sockets with GET/POST requests and responses as per issue #20 before I can complete this. Therefore I am basically needing to rework the whole codebase to do this. It is a necessary thing, however, and I think I have learned my lesson.
